app/src/main/java/com/github/swent/echo/compose/components/ListDrawer.kt

```diff
@@ -23,126 +28,254 @@ import androidx.compose.runtime.remember
+                                    listColor3.copy(alpha = 1f, green = 0.8f)
+                                else listColor3
+                            )
+                ) {
+                    Text(
+                        if (event.tags.any { tag -> tag.name.lowercase() == "epfl" }) "epfl"
+                        else if (event.tags.any { tag -> tag.name.lowercase() == "in" }) "in"
+                        else if (event.tags.any { tag -> tag.name.lowercase() == "ba6" }) "ba6"
+                        else "other",
+                        textAlign = TextAlign.Center,
+                        modifier = Modifier.align(Alignment.Center)
+                    )
+                }
+            }
         }
+        // Second floor container, only displayed when the event is selected
         if (selectedEvent.value == event.eventId) {
             Row(modifier = Modifier.testTag("list_event_details_${event.eventId}")) {
+                // Display the event description
                 Text(
                     text = event.description,
                     textAlign = TextAlign.Justify,
                     modifier =
-                        Modifier.width(200.dp).testTag("list_event_description_${event.eventId}")
+                        Modifier.width(200.dp)
+                            .padding(5.dp)
+                            .testTag("list_event_description_${event.eventId}")
                 )
                 Spacer(modifier = Modifier.width(40.dp))
-                // To add when the button will be implemented the next sprint
-                /*
                 Column {
                     val buttonWidth = 130.dp
+                    // get the context
+                    val context = LocalContext.current
+                    // To add when the button will be implemented the next Milestone
+                    /*
                     Button(
                         onClick = { },
                         modifier = Modifier.width(buttonWidth),
                     ) {
                         Text(stringResource(id = R.string.list_drawer_view_on_map))
-                    }
+                    }*/
+                    // Join event button
                     Button(
-                        onClick = { },
+                        onClick = {
+                            onJoinButtonPressed()
+                            Toast.makeText(
+                                    context,
+                                    context.getString(R.string.event_successfully_joined),
+                                    Toast.LENGTH_SHORT
+                                )
+                                .show()
+                        },
                         modifier = Modifier.width(buttonWidth),
                     ) {
                         Text(stringResource(id = R.string.list_drawer_join_event))
                     }
                 }
-
-                 */
             }
         }
     }
 }
+
```

Comment:
You can use the @ExcludeFromJacocoGeneratedReport to exclude your preview function from the unit tests coverage repoirt and increase line coverage.

Reply:
Nice tips ! Thanks